src/main/kotlin/graphics/scenery/controls/RangeSliderUI.kt

```diff
@@ -0,0 +1,550 @@
+package graphics.scenery.controls
+
+import java.awt.*
+import java.awt.event.MouseEvent
+import java.awt.geom.Ellipse2D
+import javax.swing.JComponent
+import javax.swing.JSlider
+import javax.swing.SwingUtilities
+import javax.swing.event.ChangeEvent
+import javax.swing.event.ChangeListener
+import javax.swing.plaf.basic.BasicSliderUI
+
+/*
+From https://github.com/ernieyu/Swing-range-slider/blob/master/src/slider/RangeSliderUI.java
+ */
+/**
+ * UI delegate for the RangeSlider component.  RangeSliderUI paints two thumbs,
+ * one for the lower value and one for the upper value.
+ */
+internal class RangeSliderUI
+/**
```

Comment:
This is a strange place to add a comment, please move the Kdoc above the class definition, not between the name and the constructor brackets.